src/segment-loader.js

```diff
@@ -1260,20 +1264,44 @@ export default class SegmentLoader extends videojs.EventTarget {
     // can be adjusted by the timestamp offset
     if (!segmentInfo.hasAppendedData_) {
       this.metadataQueue_.caption.push(
-        this.handleCaptions_.bind(this, simpleSegment, captions));
+        this.handleCaptions_.bind(this, simpleSegment, captionData));
       return;
     }
 
     const timestampOffset = this.sourceUpdater_.videoTimestampOffset() === null ?
       this.sourceUpdater_.audioTimestampOffset() :
       this.sourceUpdater_.videoTimestampOffset();
 
-    captions.forEach((caption) => {
-      createCaptionsTrackIfNotExists(
-        this.inbandTextTracks_, this.hls_.tech_, caption.stream);
+    const tracks = {};
+
+    // get total start/end and captions for each track/stream
+    captionData.forEach((caption) => {
+      // caption.stream is actually a track name...
+      const track = tracks[caption.stream] = tracks[caption.stream] || {
```

Comment:
`tracks[caption.stream] = tracks[caption.stream]` is this part correct?

Reply:
I cleaned up this portion, and added a comment